### PR TITLE
Feature: Implement color picker

### PR DESCRIPTION
**Pull request by FuriouZz**
Hello there, I just implemented the color picker.

I do not know why but the typing of "simple-color-picker" is not recognized.
"es-lint" is quite painful to use and do not match the VSCode formater 🤔.

---
This change is [Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/70)

### CODE REVIEW

**Comment by FuriouZz** [comment]
Capture_2021-04-12_01 28 08

File: client/web/package.json [hunk 1]
```diff
@@ -16,6 +16,8 @@
 	"license": "Apache-2.0",
 	"homepage": "https://www.graphite.design",
 	"dependencies": {
+		"lol": "github:wk-js/lol.js",
+		"simple-color-picker": "^1.0.4",
```
Comment (Keavon) [inline client/web/package.json:20]: The goal for Graphite is to avoid dependencies unless they are of significant importance, so let's aim to remove our dependence of these before merging. I would be happy to write the color picker component for you if you are concerned about losing the convenience provided by `simple-color-picker`.

File: client/web/src/App.vue [hunk 2]
```diff
@@ -18,8 +20,11 @@ html, body, #app {
 <script lang="ts">
 import { defineComponent } from "vue";
 import MainWindow from "./components/window/MainWindow.vue";
+import { NC } from "./events/NotificationCenter";
 
-const wasm = import("../wasm/pkg");
+const _wasm = import("../wasm/pkg");
```
Comment (Keavon) [inline client/web/src/App.vue:25]: Please remove the underscore.

File: client/web/src/components/panels/Document.vue [hunk 3]
```diff
@@ -6,101 +6,229 @@
...
+						:active="activeTool === 'Ellipse'"
+						@click="selectTool('Ellipse')"
+						><EllipseTool
+					/></ShelfItem>
+					<ShelfItem
+						title="Shape Tool"
+						:active="activeTool === 'Shape'"
+						@click="'tool not implemented' || selectTool('Shape')"
+						><ShapeTool
+					/></ShelfItem>
```
Comment (Keavon) [inline client/web/src/components/panels/Document.vue:228]: Please take care to not commit accidental whitespace formatting changes since it makes the code review process larger.

File: client/web/src/components/panels/Document.vue [hunk 4]
```diff
@@ -6,101 +6,229 @@
...
-						<button
-							class="secondary swatch"
-							style="background: white;"
-						></button>
-						<button
-							class="primary swatch"
-							style="background: black;"
-						></button>
-					</div>
-					<div class="swap-and-reset">
```
Comment (Keavon) [inline client/web/src/components/panels/Document.vue:95]: Swap and reset buttons should remain in the tool shelf.

File: client/web/src/components/panels/Document.vue [hunk 5]
```diff
@@ -140,34 +268,6 @@
...
-					box-sizing: unset;
-					outline: none;
-				}
-
-				.primary.swatch {
-					margin-bottom: -8px;
-				}
-			}
-
-			.swap-and-reset {
```
Comment (Keavon) [inline client/web/src/components/panels/Document.vue:168]: Swap and reset buttons should remain in the tool shelf.

File: client/web/src/components/panels/Document.vue [hunk 6]
```diff
@@ -204,8 +304,7 @@ import IconButton from "../widgets/IconButton.vue";
 import DropdownButton from "../widgets/DropdownButton.vue";
 import RadioPicker from "../widgets/RadioPicker.vue";
 import NumberInput from "../widgets/NumberInput.vue";
-import SwapButton from "../../../assets/svg/16x16-bounds-12x12-icon/swap.svg";
-import ResetColorsButton from "../../../assets/svg/16x16-bounds-12x12-icon/reset-colors.svg";
```
Comment (Keavon) [inline client/web/src/components/panels/Document.vue:208]: Swap and reset buttons should remain in the tool shelf.

File: client/web/src/components/panels/Document.vue [hunk 7]
```diff
@@ -328,25 +426,39 @@ export default defineComponent({
 			select_tool(toolName);
 		},
 		async viewModeChanged(toolIndex: number) {
-			console.log(`The view mode has been changed to index match the icon at index ${toolIndex}`);
```
Comment (Keavon) [inline client/web/src/components/panels/Document.vue:331]: Unnecessary formatting change.

File: client/web/src/components/panels/Document.vue [hunk 8]
```diff
@@ -328,25 +426,39 @@ export default defineComponent({
...
-		},
+			console.log(
+				`The view mode has been changed to index match the icon at index ${toolIndex}`
+			);
+		}
 	},
 	mounted() {
-		registerResponseHandler(ResponseType.UpdateCanvas, (responseData) => {
```
Comment (Keavon) [inline client/web/src/components/panels/Document.vue:335]: Unnecessary formatting change.

File: client/web/src/components/panels/Document.vue [hunk 9]
```diff
@@ -328,25 +426,39 @@ export default defineComponent({
...
+				.split("\n")
+				.map((shape, i) =>
+					shape.replace(
+						"#fff",
+						`#${Math.floor(
+							Math.abs(Math.sin(i + 1)) * 16777215
+						).toString(16)}`
+					)
+				)
+				.join("\n");
```
Comment (Keavon) [inline client/web/src/components/panels/Document.vue:446]: Unnecessary formatting change.

File: client/web/src/components/panels/Document.vue [hunk 10]
```diff
@@ -328,25 +426,39 @@ export default defineComponent({
...
 		});
-		registerResponseHandler(ResponseType.SetActiveTool, (responseData) => {
+		registerResponseHandler(ResponseType.SetActiveTool, responseData => {
```
Comment (Keavon) [inline client/web/src/components/panels/Document.vue:448]: I feel like you probably have a local ESLint or autoformatter config which is making these changes for you, since our ESLint config apparently isn't set up correctly. That is something I have tried to fix but wasn't successful so far. We're standardizing on always including the `(parentheses)` in arrow functions.

File: client/web/src/components/panels/Document.vue [hunk 11]
```diff
@@ -328,25 +426,39 @@ export default defineComponent({
...
 			this.activeTool = responseData;
 		});
 
 		window.addEventListener("keyup", (e: KeyboardEvent) => this.keyUp(e));
-		window.addEventListener("keydown", (e: KeyboardEvent) => this.keyDown(e));
+		window.addEventListener("keydown", (e: KeyboardEvent) =>
+			this.keyDown(e)
+		);
```
Comment (Keavon) [inline client/web/src/components/panels/Document.vue:455]: Unnecessary formatting change.

File: client/web/src/components/panels/Document.vue [hunk 12]
```diff
@@ -328,25 +426,39 @@ export default defineComponent({
...
 	},
 	data() {
 		return {
 			viewportSvg: "",
-			activeTool: "Select",
+			activeTool: "Select"
```
Comment (Keavon) [inline client/web/src/components/panels/Document.vue:460]: Please add back the closing comma here and everywhere.

File: client/web/src/lib/ColorPicker.ts [hunk 13]
```diff
@@ -0,0 +1,46 @@
+interface ColorPickerOptions {
+  el: Element;
+};
+
+export interface RGBAColor {
+  r: number;
+  g: number;
+  b: number;
+  a?: number;
```
Comment (Keavon) [inline client/web/src/lib/ColorPicker.ts:9]: `a` will always be included in every circumstance.

File: client/web/src/lib/ColorPicker.ts [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+  constructor(options: Partial<ColorPickerOptions> = {}) {
+
+  }
+
+  setColor(color: string | number | RGBAColor) {
+
+  }
+
+  getRGB() {
+    return { r: 1, g: 1, b: 1, };
```
Comment (Keavon) [inline client/web/src/lib/ColorPicker.ts:22]: Trailing comma only after multiple-line properties in an object, not in single-line objects.

File: client/web/src/types.d.ts [hunk 15]
```diff
@@ -9,3 +9,5 @@ declare module "*.svg" {
 	const component: DefineComponent;
 	export default component;
 }
+
+declare module "simple-color-picker";
```
Comment (Keavon) [inline client/web/src/types.d.ts:13]: Left over from before.

File: client/web/src/components/panels/Document.vue (hunk 10 above)
Comment (FuriouZz) [reply client/web/src/components/panels/Document.vue:405]: It seems that Vetur's prettier remove that kind of thing =/

File: client/web/src/components/panels/Document.vue (hunk 3 above)
Comment (FuriouZz) [reply client/web/src/components/panels/Document.vue]: Same thing with Vetur's formatter =/.

File: client/web/src/lib/ColorPicker.ts (hunk 13 above)
Comment (FuriouZz) [reply client/web/src/lib/ColorPicker.ts]: Updated

File: client/web/src/types.d.ts (hunk 15 above)
Comment (FuriouZz) [reply client/web/src/types.d.ts]: Removed

File: client/web/src/components/panels/Document.vue (hunk 12 above)
Comment (FuriouZz) [reply client/web/src/components/panels/Document.vue]: Vetur formatter..

File: client/web/src/components/panels/Document.vue (hunk 11 above)
Comment (FuriouZz) [reply client/web/src/components/panels/Document.vue]: Vetur formatter..

File: client/web/src/components/panels/Document.vue (hunk 9 above)
Comment (FuriouZz) [reply client/web/src/components/panels/Document.vue]: Vetur formatter..

File: client/web/src/components/panels/Document.vue (hunk 7 above)
Comment (FuriouZz) [reply client/web/src/components/panels/Document.vue]: Vetur formatter..

**Comment by Keavon** [comment]
Associated issue: https://github.com/GraphiteEditor/Graphite/issues/161

File: client/web/src/components/popovers/ColorPicker.vue [hunk 16]
```diff
@@ -115,13 +117,197 @@
...
+			},
+		},
+		saturation: {
+			rect: {
+				width: 0,
+				height: 0,
+				top: 0,
+				left: 0,
+			},
+		},
```
Comment (Keavon) [inline client/web/src/components/popovers/ColorPicker.vue:175]: Please format in a more condensed format to make these low-entropy lines take up less space:

```ts
		color: { h: 0, s: 0, v: 0, a: 1 },
		hue: {
			rect: { width: 0, height: 0, top: 0, left: 0 },
		},
		opacity: {
			rect: { width: 0, height: 0, top: 0, left: 0 },
		},
		saturation: {
			rect: { width: 0, height: 0, top: 0, left: 0 },
		},
```

File: client/web/src/components/popovers/ColorPicker.vue [hunk 17]
```diff
@@ -115,13 +117,197 @@
...
+	};
+	opacity: {
+		rect: Rect;
+	};
+}
+
+type ColorPickerState = "idle" | "move_hue" | "move_opacity" | "move_saturation";
 
 export default defineComponent({
+	colorPicker: {
```
Comment (Keavon) [inline client/web/src/components/popovers/ColorPicker.vue:145]: Is there a reason this is a custom property instead of part of the `data` section?

File: client/web/src/components/popovers/ColorPicker.vue [hunk 18]
```diff
@@ -115,13 +117,197 @@
...
+	};
+	saturation: {
+		rect: Rect;
+	};
+	opacity: {
+		rect: Rect;
+	};
+}
+
+type ColorPickerState = "idle" | "move_hue" | "move_opacity" | "move_saturation";
```
Comment (Keavon) [inline client/web/src/components/popovers/ColorPicker.vue:142]: An enum would be preferred here since it fits the pattern used in the existing codebase.

File: client/web/src/components/popovers/ColorPicker.vue [hunk 19]
```diff
@@ -115,13 +117,197 @@
...
+	hue: {
+		rect: Rect;
+	};
+	saturation: {
+		rect: Rect;
+	};
+	opacity: {
+		rect: Rect;
+	};
+}
```
Comment (Keavon) [inline client/web/src/components/popovers/ColorPicker.vue:140]: I think it's probably sufficient to define `color`, `hue`, saturation`, and `opacity` on the Vue component's `data` object without having to add the complexity of wrapping it in an interface. If there was a reason to use a custom property instead of the `data` object, disregard this.

File: client/web/src/components/popovers/ColorPicker.vue [hunk 20]
```diff
@@ -115,13 +117,197 @@
...
+			document.addEventListener("pointerup", this.onPointerUp);
+		},
+
+		removeEvents() {
+			document.removeEventListener("pointermove", this.onPointerMove);
+			document.removeEventListener("pointerup", this.onPointerUp);
+		},
+
+		onPointerDown(e: PointerEvent) {
+			const target = e.currentTarget as Element;
```
Comment (Keavon) [inline client/web/src/components/popovers/ColorPicker.vue:206]: This usage of `as` casting is okay.

File: client/web/src/components/popovers/ColorPicker.vue [hunk 21]
```diff
@@ -115,13 +117,197 @@
...
+				this.setHuePosition(e.y - colorPicker.hue.rect.top);
+			} else if (this.state === "move_opacity") {
+				this.setOpacityPosition(e.y - colorPicker.opacity.rect.top);
+			} else if (this.state === "move_saturation") {
+				this.setSaturationPosition(e.x - colorPicker.saturation.rect.left, e.y - colorPicker.saturation.rect.top);
+			}
+
+			if (this.state !== "idle") {
+				this.updateHue();
+				this.$emit("update:color", HSV2RGB(colorPicker.color));
```
Comment (Keavon) [inline client/web/src/components/popovers/ColorPicker.vue:230]: Nice! I didn't know about the `emit` feature! Perhaps after this PR you can add that approach for other widgets we use like `RadioInput.vue` which uses a weird callback-based approach.

File: client/web/src/components/popovers/ColorPicker.vue [hunk 22]
```diff
@@ -115,13 +117,197 @@
...
+			if (this.state !== "idle") {
+				this.state = "idle";
+				this.removeEvents();
+			}
+		},
+
+		updateRects() {
+			const { colorPicker } = this.$options as { colorPicker: ColorPickerData };
+
+			const saturationPicker = this.$refs.saturationPicker as HTMLDivElement;
```
Comment (Keavon) [inline client/web/src/components/popovers/ColorPicker.vue:244]: This type of `as` casting is as fine.

File: client/web/src/components/popovers/ColorPicker.vue [hunk 23]
```diff
@@ -115,13 +117,197 @@
...
+			const huePosition = clamp(y, 0, colorPicker.hue.rect.height);
+			hueCursor.style.transform = `matrix(1, 0, 0, 1, 0, ${huePosition})`;
+			colorPicker.color.h = clamp(1 - huePosition / colorPicker.hue.rect.height);
+		},
+
+		setOpacityPosition(y: number) {
+			const { colorPicker } = this.$options as { colorPicker: ColorPickerData };
+			const opacityCursor = this.$refs.opacityCursor as HTMLDivElement;
+			const opacityPosition = clamp(y, 0, colorPicker.opacity.rect.height);
+			opacityCursor.style.transform = `matrix(1, 0, 0, 1, 0, ${opacityPosition})`;
```
Comment (Keavon) [inline client/web/src/components/popovers/ColorPicker.vue:287]: Why a `matrix`? Wouldn't `translate` be sufficient? It would be somewhat more readable.

File: client/web/src/components/popovers/ColorPicker.vue [hunk 24]
```diff
@@ -115,13 +117,197 @@
...
+			const hueCursor = this.$refs.hueCursor as HTMLDivElement;
+			const huePosition = clamp(y, 0, colorPicker.hue.rect.height);
+			hueCursor.style.transform = `matrix(1, 0, 0, 1, 0, ${huePosition})`;
+			colorPicker.color.h = clamp(1 - huePosition / colorPicker.hue.rect.height);
+		},
+
+		setOpacityPosition(y: number) {
+			const { colorPicker } = this.$options as { colorPicker: ColorPickerData };
+			const opacityCursor = this.$refs.opacityCursor as HTMLDivElement;
+			const opacityPosition = clamp(y, 0, colorPicker.opacity.rect.height);
```
Comment (Keavon) [inline client/web/src/components/popovers/ColorPicker.vue:286]: It might be cleaner to use percentages instead of pixels. But that's not a hard requirement, just a suggestion.

File: client/web/src/components/popovers/ColorPicker.vue (hunk 17 above)
Comment (FuriouZz) [reply client/web/src/components/popovers/ColorPicker.vue:145]: This data does not need to be reactive.

File: client/web/src/components/popovers/ColorPicker.vue (hunk 19 above)
Comment (FuriouZz) [reply client/web/src/components/popovers/ColorPicker.vue:140]: Data is for reactive values, this is the way in Vue to add non-reactive data to a component

File: client/web/src/components/widgets/inputs/SwatchPairInput.vue [hunk 25]
```diff
@@ -28,6 +28,7 @@
 		position: relative;
 
 		button {
+			--swatch-color: white;
```
Comment (Keavon) [inline client/web/src/components/widgets/inputs/SwatchPairInput.vue:31]: If you don't mind switching this to `#ffffff`, it would be preferred over the name `white`.

File: client/web/src/components/widgets/inputs/SwatchPairInput.vue [hunk 26]
```diff
@@ -71,11 +90,47 @@ export default defineComponent({
...
+			button.style.setProperty("--swatch-color", `rgba(${color.r}, ${color.g}, ${color.b}, ${color.a})`);
+
+			color = RGB2Floats(this.secondaryColor);
+			update_secondary_color(new Color(color.r, color.g, color.b, color.a));
+		},
 	},
 	data() {
 		return {
 			PopoverDirection,
+			primaryColor: {
```
Comment (Keavon) [inline client/web/src/components/widgets/inputs/SwatchPairInput.vue:117]: Please reverse these (primary is black, secondary is white) and combine them onto one line each (two lines total).

File: client/web/src/components/popovers/ColorPicker.vue (hunk 19 above)
Comment (FuriouZz) [reply client/web/src/components/popovers/ColorPicker.vue:140]: https://vuejs.org/v2/api/#vm-options

File: client/web/src/components/popovers/ColorPicker.vue (hunk 17 above)
Comment (FuriouZz) [reply client/web/src/components/popovers/ColorPicker.vue:145]: https://vuejs.org/v2/api/#vm-options

File: client/web/src/lib/utils.ts [hunk 27]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+export interface HSV {
+	h: number;
+	s: number;
+	v: number;
+	a: number;
+}
+
+export function HSV2RGB(hsv: HSV): RGB {
```
Comment (Keavon) [inline client/web/src/lib/utils.ts:15]: Please rename to `hsvToRgb`.

File: client/web/src/lib/utils.ts [hunk 28]
```diff
@@ -0,0 +1,68 @@
...
+	const q = v * (1 - f * s);
+	const t = v * (1 - (1 - f) * s);
+	const mod = i % 6;
+	const r = Math.round([v, q, p, p, t, v][mod]);
+	const g = Math.round([t, v, v, q, p, p][mod]);
+	const b = Math.round([p, p, t, v, v, q][mod]);
+	return { r, g, b, a: hsv.a };
+}
+
+export function RGB2HSV(rgb: RGB) {
```
Comment (Keavon) [inline client/web/src/lib/utils.ts:31]: Please rename to `rgbToHsv`.

File: client/web/src/lib/utils.ts [hunk 29]
```diff
@@ -0,0 +1,68 @@
...
+				h = (r - g) / d + 4;
+				break;
+			default:
+		}
+		h /= 6;
+	}
+	return { h, s, v, a: rgb.a };
+}
+
+export function RGB2Floats(rgb: RGB) {
```
Comment (Keavon) [inline client/web/src/lib/utils.ts:59]: Please rename to `rgbToDecimalRgb` and add a return type.

File: client/web/src/components/panels/Document.vue [hunk 30]
```diff
@@ -319,9 +315,6 @@ export default defineComponent({
 
 		window.addEventListener("keyup", (e: KeyboardEvent) => this.keyUp(e));
 		window.addEventListener("keydown", (e: KeyboardEvent) => this.keyDown(e));
-
-		// TODO: Implement an actual UI for chosing colors (this is completely temporary)
-		this.updatePrimaryColor({ r: 247 / 255, g: 76 / 255, b: 0 / 255, a: 0.6 });
 	},
```
Comment (Keavon) [inline client/web/src/components/panels/Document.vue:318]:
```scss
.viewport {
			flex: 1 1 100%;
			.canvas {
				background: var(--color-1-nearblack);
				width: 100%;
				height: 100%;
				svg {
					width: 100%;
					height: 100%;
				}
```

We can add `background: #ffffff;` in order to make the SVG canvas white so our default black primary color shapes will be visible.

File: client/web/src/components/popovers/ColorPicker.vue (hunk 17 above)
Comment (FuriouZz) [reply client/web/src/components/popovers/ColorPicker.vue:145]: Alternatively, we can put the object in data() and prefix with an `_`, VueJS does not proxify object with underscore. 
https://v3.vuejs.org/api/options-data.html#data-2

File: client/web/src/components/popovers/ColorPicker.vue (hunk 19 above)
Comment (FuriouZz) [reply client/web/src/components/popovers/ColorPicker.vue:140]: Alternatively, we can put the object in data() and prefix with an _, VueJS does not proxify object with underscore.
https://v3.vuejs.org/api/options-data.html#data-2

File: client/web/src/components/popovers/ColorPicker.vue [hunk 31]
```diff
@@ -115,13 +117,197 @@
...
+		},
+		saturation: {
+			rect: {
+				width: 0,
+				height: 0,
+				top: 0,
+				left: 0,
+			},
+		},
+	} as ColorPickerData,
```
Comment (Keavon) [inline client/web/src/components/popovers/ColorPicker.vue:176]: It would be great if we can avoid unsafe casting (since `as` tells TypeScript to ignore type checks and blindly believe the code that the type cast is correct). When we need to cast something to an HTML element, that's fine of course since there is no other way around it. This applies throughout the PR in a bunch of places.

File: client/web/src/components/panels/Document.vue [hunk 32]
```diff
@@ -149,7 +149,7 @@
 			flex: 1 1 100%;
 
 			.canvas {
-				background: var(--color-1-nearblack);
```
Comment (Keavon) [inline client/web/src/components/panels/Document.vue:152]: Could you specifically apply the `background: #ffffff;` to the `svg` tag below, and restore `background: var(--color-1-nearblack);` on the `.canvas`?

File: client/web/src/components/popovers/ColorPicker.vue [hunk 33]
```diff
@@ -202,60 +172,75 @@ export default defineComponent({
...
+			return this.$refs[name] as T;
+		},
 
-			if (picker && this.state === "idle") {
+		onPointerDown(e: PointerEvent) {
+			if (!(e.currentTarget instanceof Element)) return;
+			const picker = e.currentTarget.getAttribute("data-picker-action");
+			this.state = (() => {
+				switch (picker) {
+					case "MoveHue":
```
Comment (Keavon) [inline client/web/src/components/popovers/ColorPicker.vue:184]: case ColorPickerState.MoveHue:

Comment (Keavon) [reply client/web/src/components/popovers/ColorPicker.vue:184]: Sorry, disregard.

File: client/web/src/components/popovers/ColorPicker.vue [hunk 34]
```diff
@@ -202,60 +172,75 @@ export default defineComponent({
 			document.removeEventListener("pointerup", this.onPointerUp);
 		},
 
-		onPointerDown(e: PointerEvent) {
-			const target = e.currentTarget as Element;
-			const picker = target.getAttribute("data-picker-type");
+		getRef<T>(name: string) {
+			return this.$refs[name] as T;
+		},
```
Comment (Keavon) [inline client/web/src/components/popovers/ColorPicker.vue:177]: Neat trick! I love this.

File: client/web/src/components/widgets/inputs/SwatchPairInput.vue [hunk 35]
```diff
@@ -37,6 +38,21 @@
...
+			background-position: 0 0, 8px 8px;
+
+			&::before {
+				content: "";
+				display: block;
+				width: 100%;
+				height: 100%;
+				background: var(--swatch-color);
+				background-size: 16px 16px;
+				background-position: 0 0, 8px 8px;
```
Comment (Keavon) [inline client/web/src/components/widgets/inputs/SwatchPairInput.vue:53]: I don't think these are used for anything. I was able to delete them and didn't see any changes. Correct me if I'm wrong though!

File: client/web/src/components/widgets/inputs/SwatchPairInput.vue [hunk 36]
```diff
@@ -37,6 +38,21 @@
...
+				border-radius: 50%;
```
Comment (Keavon) [inline client/web/src/components/widgets/inputs/SwatchPairInput.vue:54]: To avoid any possible antialiasing problems, it is probably slightly more robust to get rid of this border-radius and add `overflow: hidden;` to the parent `button`.

File: client/web/src/components/popovers/ColorPicker.vue [hunk 37]
```diff
@@ -46,7 +46,7 @@
 	}
 
 	.opacity-picker {
-		background: linear-gradient(to bottom, #ff0000, transparent);
+		background: linear-gradient(to bottom, var(--hue), transparent);
```
Comment (Keavon) [inline client/web/src/components/popovers/ColorPicker.vue:49]: We want the final color here, not the hue.

**Comment by Keavon** [comment]
Awesome work! Just a few small changes and then it'll be time to merge. Be sure to rebase also so it's up to date with the current state of master.

File: client/web/src/components/popovers/ColorPicker.vue [hunk 38]
```diff
@@ -14,13 +14,14 @@
 
 <style lang="scss">
 .popover-color-picker {
-	--hue: #ff0000;
+	--hue-picker-color: #ff0000;
```
Comment (Keavon) [inline client/web/src/components/popovers/ColorPicker.vue:17]: Shouldn't this say `--saturation-picker-color`?

Comment (Keavon) [reply client/web/src/components/popovers/ColorPicker.vue]: Looks good! As a small nitpick, would you mind changing `--saturation-picker-color` to `--saturation-picker-hue` which is technically more accurate? You probably didn't see my edit to the above description when I added the second sentence. I'll go ahead and hit approve now and you're ready to merge!

**Review by Keavon — APPROVED** [review]

**Review by Keavon — APPROVED** [review]
> Reviewed 1 of 8 files at r1.
> *[Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/70)* status: 0 of 16 files reviewed, 31 unresolved discussions (waiting on @FuriouZz and @Keavon)

**Review by Keavon — APPROVED** [review]
> Reviewed 1 of 12 files at r3, 4 of 12 files at r6, 1 of 6 files at r7, 1 of 4 files at r8, 1 of 3 files at r9, 1 of 1 files at r10, 8 of 8 files at r11.
> *[Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/70)* status: all files reviewed, 31 unresolved discussions (waiting on @FuriouZz and @Keavon)
